### DISCUSSION TITLE
Cannot use eth0 as mgt interface without using a static route.

### DISCUSSION THREAD

**Discussion by GeneralKuruma** [wifinigel/wiperf#70] [Bugs - I think I may have found a bug]
Hello,

I would like to send my wiperf test results over an ethernet interface, but whenever wiperf runs it adds a metric 500 to my eth0 interface. This shuts down my SSH connection and makes it so it can't actually send the data over that interface to a splunk database. (basically makes it so i can't use the eth0 connection):

> 2021-11-25 13:39:15,790 - Probe_Log - INFO - ping_host: google.com, pkts_tx: 1, pkts_rx: 1, pkt_loss: 0, rtt_avg: 7.362
> 2021-11-25 13:39:15,806 - Probe_Log - INFO -   Checked interface route to : xx.xxx.xxx.xx. Result: xx.xxx.xxx.xx via xx.xxx.xxx.xx dev eth0 src xx.xxx.xxx.xx uid 0
> 2021-11-25 13:39:15,807 - Probe_Log - WARNING -   We are not using the interface required to perform our tests due to a routing issue in this unit - attempt route addition to fix issue
> 2021-11-25 13:39:15,809 - Probe_Log - INFO -   DNS hostname lookup : google.com. Result: xx.xxx.xxx.xx
> 2021-11-25 13:39:15,821 - Probe_Log - INFO -   Checked interface route to : xx.xxx.xxx.xx. Result: default via xx.xxx.xxx.xx dev eth0
> 2021-11-25 13:39:15,833 - Probe_Log - INFO -   [Route Injection] Deleting route: default via xx.xxx.xxx.xx dev eth0
> 2021-11-25 13:39:15,846 - Probe_Log - INFO -   [Route Injection] Re-adding deleted route with new metric: default via xx.xxx.xxx.xx dev eth0 metric 500
> 2021-11-25 13:39:15,847 - Probe_Log - INFO -   [Route Injection] Checking probe mode: 'wireless'
> 2021-11-25 13:39:15,858 - Probe_Log - INFO -   [Route Injection] Adding new route: default dev wlan0
> 2021-11-25 13:39:15,859 - Probe_Log - INFO -   [Route Injection] Route injection complete
> 2021-11-25 13:39:15,859 - Probe_Log - DEBUG - Bouncing interface (platform type = rpi)
> 2021-11-25 13:39:15,859 - Probe_Log - INFO - Bouncing interface wlan0 (platform type = rpi)
> 2021-11-25 13:39:15,859 - Probe_Log - WARNING - Taking interface down...
> 2021-11-25 13:39:28,746 - Probe_Log - WARNING - Bringing interface up...
> 2021-11-25 13:39:53,425 - Probe_Log - INFO - Interface bounce completed OK.
> 2021-11-25 13:39:53,426 - Probe_Log - INFO -   Checking if route injection worked...
> 2021-11-25 13:39:53,437 - Probe_Log - INFO -   Checked interface route to : xx.xxx.xxx.xx. Result: xx.xxx.xxx.xx via xx.xxx.xxx.xx dev wlan0 src xx.xxx.xxx.xx uid 0
> 2021-11-25 13:39:53,438 - Probe_Log - INFO -   Routing issue corrected OK.
> 2021-11-25 13:39:53,439 - Probe_Log - INFO - Checking we can get to the management platform (host = xxxxx.xx.xxxxxx.xxxx.xxx, port = xxxx, type = splunk)
> 2021-11-25 13:39:58,541 - Probe_Log - INFO -   DNS hostname lookup : xxxxx.xx.xxxxxx.xxxx.xxx. Result: xx.xxx.xxx.xx
> 2021-11-25 13:39:58,544 - Probe_Log - INFO -   Checking we will send mgt traffic over configured interface 'eth0' mode.
> 2021-11-25 13:39:58,571 - Probe_Log - INFO -   Checked interface route to : xx.xxx.xxx.xx. Result: xx.xxx.xxx.xx via xx.xxx.xxx.xx dev wlan0 src xx.xxx.xxx.xx uid 0
> 2021-11-25 13:39:58,573 - Probe_Log - INFO -   Mgt interface will be routed over wrong interface: xx.xxx.xxx.xx via xx.xxx.xxx.xx dev wlan0 src xx.xxx.xxx.xx uid 0
> 2021-11-25 13:39:58,573 - Probe_Log - WARNING -   We are not using the interface required for mgt traffic due to a routing issue in this unit - attempt route addition to fix issue
> 2021-11-25 13:39:58,574 - Probe_Log - INFO -   [Route Injection] Attempting static route insertion to fix routing issue
> 2021-11-25 13:39:58,599 - Probe_Log - INFO -   [Route Injection] Adding new mgt traffic route: xxxxx.xx.xxxxxx.xxxx.xxx dev eth0
> 2021-11-25 13:39:58,600 - Probe_Log - INFO -   [Route Injection] Route injection (mgt)complete
> 2021-11-25 13:39:58,601 - Probe_Log - INFO -   Checking if route injection worked...
> 2021-11-25 13:39:58,614 - Probe_Log - INFO -   DNS hostname lookup : xxxxx.xx.xxxxxx.xxxx.xxx. Result: xx.xxx.xxx.xx
> 2021-11-25 13:39:58,614 - Probe_Log - INFO -   Checking we will send mgt traffic over configured interface 'eth0' mode.
> 2021-11-25 13:39:58,641 - Probe_Log - INFO -   Checked interface route to : xx.xxx.xxx.xxx. Result: xx.xxx.xxx.xx via xx.xxx.xxx.xx dev wlan0 src xx.xxx.xxx.xx uid 0
> 2021-11-25 13:39:58,642 - Probe_Log - INFO -   Mgt interface will be routed over wrong interface: xx.xxx.xxx.xx via xx.xxx.xxx.xx dev wlan0 src xx.xxx.xxx.xx uid 0
> 2021-11-25 13:39:58,642 - Probe_Log - WARNING -   We still have a routing issue. Will have to exit as mgt traffic over correct interface not possible
> 2021-11-25 13:39:58,643 - Probe_Log - WARNING -   Suggest making static routing additions or adding an additional metric to the interface causing the issue.
> 2021-11-25 13:39:58,643 - Probe_Log - WARNING -   (*** Note ***: check you have configured the correct mgt interface if this message persists)

I fixed this problem by adding static routes, however when i connect the probe on a different subnet i will always have to change the static routes (obviously). My main question is: Is it possible to still have eth0 as a mgt interface (to send the tests to the splunk database) without having to use static routes? Basically in the end i want to be able to connect the probe and have it connect automatically to a subnet on eth0 instead of having to change the static route manually every different subnet.

Personally i love using Wiperf and want to thank you for all your work, any help would be greatly appreciated.
Regards
Brent

**Comment by wifinigel** [comment]
Hi Brent,

This issue is usually caused by the incorrect value being set in the configuration file for the interface to be used for management.

Getting the routing right is quite tricky, but wiperf should take care of things for you  if you set the interfaces correctly in the config file.

Can you please check if your config file is set as follows and that you have no static routes set please:

```
[General]
; global test mode: 'wireless' or 'ethernet'
; 
; wireless mode: 
;    - test traffic runs over wireless interface
;    - management traffic (i.e. result data) sent over interface specified in mgt_if parameter
; ethernet mode:
;    - test traffic runs over ethernet interface
;    - management traffic (i.e. result data) sent over interface specified in mgt_if parameter
;
**probe_mode: wireless**
;
; ------------- ethernet mode parameters ------------
; eth interface name - set this as per the output of an ifconfig command (usually eth0)
**eth_if: eth0**
; ---------------------------------------------------
;
; ------------- wireless mode parameters ------------
; wlan interface name - set this as per the output of an iwconfig command (usually wlan0)
**wlan_if: wlan0**
; ---------------------------------------------------
;
; -------------mgt interface parameters ------------
; interface name over which mgt traffic is sent (i.e. how we get to our management
; server) - options: wlan0, eth0, ztxxxxxx (ZeroTier), lo (local instance of Influx)
**mgt_if: eth0**
```
If this fails to fix things, can you please send me a copy of your config.ini file and the output of the commands "ifconfig -a" and "ip route" while in the failed state and no static routes manually add please so that I can analyse what is going on (wifinigel@gmail.com)

Also, are you using the most recent version of wiperf?

Regards

Nigel.

**Reply by GeneralKuruma** [reply]
Hi Nigel.

First of all, thank for answering so quickly!
I've sent you a mail regarding the config.ini file and output of the commands ipconfig-a and ip route since the fix did not work.

Regards

Brent.

**Reply by wifinigel** [reply]
Hi Brent,

Yes I have the email, I will try and take a look at this issue this week for you. I'll be interested to see what's going on here :)

Nigel.
